File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int>result;
        unordered_map<int,int>mp;
        priority_queue<pair<int,int>>pq;
        for(auto elem:nums)
        {
            mp[elem]++;
        }
        for(auto i:mp)
        {
            pq.push({i.second,i.first});
        }
        while(!pq.empty() && k--)
        {
            result.push_back(pq.top().second);
            pq.pop();
        }
        return result;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic=Counter(nums)
        print(dic)
        mat=sorted([[key,val] for key,val in dic.items()],key=lambda x:x[1])
        print(mat)
        l=[]
        for i in range(k):
            x=mat.pop()
            l.append(x[0])
           
        print(l)
        return l